### DISCUSSION TITLE
About migration error of restored Byron Wallet

### DISCUSSION THREAD

**Discussion by zktky** [CardanoSolutions/ByronWalletRecovery#2] [Tell Your Story]
Hi.
I also want to participate in this project.
I also have an old Byron Wallet and am having trouble transferring funds.

I don't know if it will be helpful, but I used the [Cardano Wallet Backend API (v2022-07-01) ](https://input-output-hk.github.io/cardano-wallet/api/edge/).
I would like to share the error details in the transfer of funds.

The wallet restored with sercret.key from the latest version (v4.11.0) from daedlus has the balance restored normally and is also displayed.
I get an error when trying to move to a new shelly wallet.

The error log is as follows.

```
{
    "code": "created_invalid_transaction",
    "message": "The submitted transaction was rejected by the local node. Here's an error message that may help with debugging: TxValidationErrorInMode (ShelleyTxValidationError ShelleyBasedEraAlonzo (ApplyTxError [UtxowFailure (WrappedShelleyEraFailure (MissingVKeyWitnessesUTXOW (WitHashes (fromList [KeyHash \"3cXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX1\",KeyHash \"ffXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX2\"]))))])) AlonzoEraInCardanoMode"
}
```

I wondered if there was a problem with the transmission logic in the daedlus application, so I tried to transfer funds in the same way using the Cardano Wallet Backend API.
"Create a migration plan" worked fine, but "migrations" gives the same error as the app.

■ API executed

- Create a migration plan ： https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/createByronWalletMigrationPlan
- Migrate ： https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/migrateByronWallet

■ Execution method

```
curl -X POST -H "Content-Type: application/json" --cacert "$HOME/Library/Application Support/Daedalus Mainnet/tls/client/ca.crt" \
  --cert "$HOME/Library/Application Support/Daedalus Mainnet/tls/client/client.pem" \
  -d '{"passphrase": "{The wallet's master passphrase.}": ["{The recipient addresses.}"]}' \
  https://localhost:{daedlus_local_port}/v2/byron-wallets/{walletId}/migrations/plan 
```

```
curl -X POST -H "Content-Type: application/json" --cacert "$HOME/Library/Application Support/Daedalus Mainnet/tls/client/ca.crt" \
  --cert "$HOME/Library/Application Support/Daedalus Mainnet/tls/client/client.pem" \
  -d '{"passphrase": "{The wallet's master passphrase.}","addresses": ["{The recipient addresses.}"]}' \
  https://localhost:{daedlus_local_port}/v2/byron-wallets/{walletId}/migrations/ 
```

■ Concerns

"Create a migration plan" worked fine and received a Response,
The contents of Response are as follows. I'm worried that the assets in the selections element are not included,
I can't tell if this is a normal data structure.

-  Balance_selected ・ ・ ・ Quantity contains all funds
-  Total_fee ・ ・ ・ Quantity is calculated normally
-  Selections ・ ・ ・ There are multiple elements (elements with 10 different ids in inputs), and the quantity according to the funds is normally entered from amount.quantity.However, assets are empty in all elements. policy_id included in assets has no element
-  Balance_leftover ・ ・ ・ quantity: 0.

---

I still have a lot of money left in Byron Wallet, so I'm willing to help with this project.
Please let me know if you have a good investigation method or solution. Also, let me discuss it positively.

Thank you.

**Comment by Lovely-Thing** [comment]
I'd like to fix the problems.

**Comment by zktky** [comment]
@Lovely-Thing 

thank you for your reply. There was information that I could provide. I will cooperate generously. Also, please let me know if there is anything I can do. I can also use APIs and development tools.
